compiler+runtime/src/cpp/clojure/string_native.cpp

```diff
@@ -193,4 +193,106 @@ namespace clojure::string_native
     auto const pos(try_object<obj::integer>(from_index)->data);
     return static_cast<i64>(s_str.rfind(value_str, pos));
   }
+
+  static jtl::immutable_string::size_type triml_index(jtl::immutable_string const &s)
+  {
+    auto const s_size(s.size());
+    jtl::immutable_string::size_type i{ 0 };
+
+    for(; i < s_size; i++)
+    {
+      if(!std::isspace(s[i]))
+      {
+        break;
+      }
+    }
+
+    return i;
+  }
+
+  object_ref triml(object_ref const s)
+  {
+    auto const s_str(runtime::to_string(s));
+    auto const l(triml_index(s_str));
+
+    if(l == 0)
+    {
+      return s;
+    }
+
+    return make_box(s_str.substr(l));
+  }
+
+  static jtl::immutable_string::size_type trimr_index(jtl::immutable_string const &s)
+  {
+    auto const s_size(s.size());
+    jtl::immutable_string::size_type i{ s_size };
+
+    for(; i > 0; i--)
```

Comment:
It's always good practice, in C++, to use pre-increment/decrement, rather than post. This is because those operators are overloadable and values like iterators may provide them. Using the post version will require creating a copy of that value, compared to the pre version, which updates in place. So, this can become a performance concern, since iteration is basically always done in loops (where perf matters most).

All this to say, it's a red flag to see post usage, since it either means complex logic is happening, there's a perf concern, or somebody's not following the rules. :joy: 

So, while it won't affect the perf here, please change it to `--i` (same in the other fns) for both consistency and sanity.

Reply:
Good to know, thanks for the breakdown!